docs/develop/typescript/cancellation.mdx

```diff
@@ -383,3 +383,20 @@ export async function shieldAwaitedInRootScope(): Promise<any> {
 ```
 
 <!--SNIPEND-->
+
+## Cancel an Activity from a Workflow {#cancel-an-activity}
+
+Canceling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout.
```

Comment:
Link to Heartbeat recommended, plus a sentence or so along the lines that a Heartbeat is a proof-of-life timeout used for long-running Activities.  
This sentence also parallels 391 quite a lot. This is a good place once your thoughts are down to step back and consider wording for clarity. (Oh, I see Angela got there first. Thanks, Angela!)

Reply:
@fairlydurable I only moved that subsection out from the Failure Detection page. I agree it needs rework, but if you don't mind, I would prefer to consider that one out of scope for this PR.

Reply:
I haven't looked at what's already written on the .Net Cancellation page; when we get there (either I or somebody else), the .Net content should probably be the reference.

Reply:
You're doing something awesome. We appreciate it.